MapboxCoreNavigation/RouteController.swift

```diff
@@ -472,3 +618,134 @@ extension RouteController: CLLocationManagerDelegate {
         incrementRouteProgress(alertLevel, location: location, updateStepIndex: updateStepIndex)
     }
 }
+
+struct SessionState {
+    let identifier = UUID()
+    var departureTimestamp: Date?
+    var arrivalTimestamp: Date?
+    
+    var totalDistanceCompleted: CLLocationDistance = 0
+    
+    var numberOfReroutes = 0
+    var lastReroute: Date?
```

Comment:
`lastRerouteDate` would be clearer; otherwise, one could reasonably see `lastReroute` somewhere in the code and think it refers to a coordinate.